core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -260,9 +260,12 @@ private[spark] class SparkSubmit extends Logging {
           case m if m.startsWith("spark") => STANDALONE
           case m if m.startsWith("k8s") => KUBERNETES
           case m if m.startsWith("local") => LOCAL
-          case _ =>
-            error("Master must either be yarn or start with spark, k8s, or local")
-            -1
+          case _ => EXTERNAL
+//          case _ =>
+//            error("Master must either be yarn or start with spark, k8s, or local, or " +
+//              "in case of external cluster, " +
+//              "the master URL must be handled by one external cluster manager.")
+//            -1
```

Comment:
The external cluster manager should be asked here whether it supports the given master url.

Reply:
@EnricoMi we cannot ask at this point as we need to load the JARs first which contain the cluster manager, and that happens later in the code. This is why I delegated the check to the end.

Reply:
Where in the code are the jars loaded? Can you point me to it?